File: 125-ValidPalindrome/125-ValidPalindrome.js
// Last updated: 26/12/2025, 16:01:54
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.toLowerCase().replace(/[^a-z0-9]/g, '');

    let start = 0;
    let end = s.length-1;

    while(end > start){
        if(s.charAt(start).toLowerCase() != s.charAt(end).toLowerCase()){
            return false
        }
        start++
        end--
    }

    return true
};

File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 26/12/2025, 16:01:57
class Solution {
    public boolean isPalindrome(String s) {        
        s = s.toLowerCase().replaceAll("[^a-z0-9]","");
       
        return s.equals(String.valueOf(new StringBuilder(s).reverse()));
    }
}